src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ThrowableToErrorMapper.java

```diff
@@ -70,10 +71,40 @@ public final class ThrowableToErrorMapper {
             message = "Mismatched vector dimension";
           }
           return new CommandResult.Error(message, fieldsForMetricsTag, fields, Response.Status.OK);
-        } else if (throwable instanceof NodeUnavailableException
-            || throwable instanceof DriverException
-            || throwable instanceof AllNodesFailedException
-            || throwable instanceof NoNodeAvailableException) {
+        } else if (throwable instanceof DriverException) {
+          if (throwable instanceof AllNodesFailedException) {
+            Map<Node, List<Throwable>> nodewiseErrors =
+                ((AllNodesFailedException) throwable).getAllErrors();
+            if (!nodewiseErrors.isEmpty()) {
+              List<Throwable> errors = nodewiseErrors.values().iterator().next();
+              if (errors != null && !errors.isEmpty()) {
+                Throwable error =
+                    errors.stream()
+                        .findAny()
+                        .filter(
+                            t ->
+                                t instanceof AuthenticationException
+                                    || t instanceof IllegalArgumentException)
+                        .orElse(null);
+                // connecting to oss cassandra throws AuthenticationException for invalid
+                // credentials connecting to AstraDB throws IllegalArgumentException for invalid
+                // token/credentials
+                if (error instanceof AuthenticationException
+                    || (error instanceof IllegalArgumentException
```

Comment:
Oh, So IllegalArgumentException is the issue.

Reply:
Yes, correct.